adapters/ratalert/index.js

```diff
@@ -0,0 +1,110 @@
+require("dotenv").config();
+
+const moment = require("moment");
+const BigNumber = require("bignumber.js");
+const Matic = require("../../sdk/matic");
+const axios = require("axios");
+const URL = "http://nft-sales-service.dappradar.com/open-source";
+const KEY = process.env.DAPPRADAR_API_KEY;
+const path = require("path");
+
+class RatAlert {
+    constructor() {
+        this.name = "ratalert";
+        this.symbol = "ETH";
+        this.token = "0x7ceb23fd6bc0add59e62ac25578270cff1b9f619"; // WETH
+        this.protocol = "polygon";
+        this.block = 28184455;
+        this.contract = "0x3242CCF5d7f35BaB5A560e7ae2Fa64EfDCa9Cd1c"; // RATCAST
+        this.events = ["Transfer"];
```

Comment:
We are automatically catching mints. So if you feel that this provider doesn't do anything else, but track mints - I'll close the PR.

Reply:
Well what it does is get the correct price for sell transactions on OpenSea. It will skip Transfer events with a zero value or anything else than WETH. Mints were MATIC & now a native token, these are skipped in line 80. Not sure if this needs to be changed then. Please advise. ;)

Reply:
Can you send me some transactions where mints were made via different tokens and I'll let you know if we are already tracking that kind of sales/mints.

Reply:
I guess the challenge in our case is that a mint tx always involves 2 transactions:

1. The mint request from the user that includes the payment, e.g with [FFOOD](https://polygonscan.com/tx/0xbfe4816567e72e023fccf56f436fa36dea233847f9696817867a1498dfbc6be0) & [MATIC](https://polygonscan.com/tx/0x44488de0e00eb51c0d4c5fa2352513c56d35201a0bdabbc68e56a2f8c563c170)

And, since we use VRF,

2. The fulfillment coming from ChainLink that delivers the randomness for the actual mint, e.g. [this one](https://polygonscan.com/tx/0x4014f34ea782843995ad4f8642d5c050083c5724f66e779e1f79cc4f08aae242)

Reply:
I see a few issues here:
1. According to your code, you'll never get any entities returned, because you're returning nothing if transaction value is zero, but with the transfer event that you've sent me it's always zero.
2. You'll never know how much you paid for the specific NFT, unless there's a hard-coded value in the contract (I didn't check it myself)

Reply:
Let me first thank you for taking the time to work through this with me, it's much appreciated, @liutaurasjanuskas.

You're right, my code will not catch mints. It's intended to track OpenSea purchases like [this one](https://polygonscan.com/tx/0x9abf6b958b648724518c89edf7cdd54de9fa21319b52de1ea2f2796c1576c137).

Regarding mints, our VRF setup is not the only challenge. Another is our dynamic mint price: Gen0 tokens were paid with $MATIC but the $FFOOD price for Gen1 goes up the more NFTs are minted. That's why there is no hard-coded value in the contract. What I could do is hard-code the values for mint Transfer events in the adapter, wdyt?

Reply:
@juggernod We are currently tracking Opensea and its sales. I've just checked and we have the sale, that you've sent to me as an example, in our database.

Also, I've noticed that your collections doesn't have a dapp page created. You can submit your dapp here: https://dappradar.com/submit-dapp

Let me know if something is unclear and I'll try to answer those questions.

Reply:
Hello @juggernod, I might have missed something here. Is this resolved or not?